pixeltable/tool/create_test_db_dump.py

```diff
@@ -141,41 +123,124 @@ def create_tables(self) -> None:
                 'c5': c5_data[i],
                 'c6': c6_data[i],
                 'c7': c7_data[i],
+                'c8': None
             }
             for i in range(num_rows)
         ]
+
+        self.__add_expr_columns(t, 'base_table')
         t.insert(rows)
+
         pxt.create_dir('views')
-        v = pxt.create_view('views.sample_view', t, filter=(t.c2 < 50))
-        _ = pxt.create_view('views.sample_snapshot', t, filter=(t.c2 >= 75), is_snapshot=True)
+
+        # simple view
+        v = pxt.create_view('views.view', t, filter=(t.c2 < 50))
+        self.__add_expr_columns(v, 'view')
+
+        # snapshot
+        _ = pxt.create_view('views.snapshot', t, filter=(t.c2 >= 75), is_snapshot=True)
+
+        # view of views
+        vv = pxt.create_view('views.view_of_views', v, filter=(t.c2 >= 25))
+        self.__add_expr_columns(vv, 'view_of_views')
+
+        # empty view
         e = pxt.create_view('views.empty_view', t, filter=t.c2 == 4171780)
         assert e.count() == 0
-        # Computed column using a library function
-        v['str_format'] = pxt.functions.string.str_format('{0} {key}', t.c1, key=t.c1)
-        # Computed column using a bespoke stored udf
-        v['test_udf'] = test_udf_stored(t.c2)
-        # Computed column using a batched function
-        # (apply this to the empty view, since it's a "heavyweight" function)
-        e['batched'] = pxt.functions.huggingface.clip_text(t.c1, model_id='openai/clip-vit-base-patch32')
-        # computed column using a stored batched function
-        v['test_udf_batched'] = test_udf_stored_batched(t.c1, upper=False)
-        # astype
-        v['astype'] = t.c1.astype(pxt.FloatType())
+        self.__add_expr_columns(e, 'empty_view', include_expensive_functions=True)
 
         # Add remotes
         from pixeltable.datatransfer.remote import MockRemote
-        v.link_remote(
+        v._link(
             MockRemote({'int_field': pxt.IntType()}, {'str_field': pxt.StringType()}),
-            col_mapping={'test_udf': 'int_field', 'c1': 'str_field'}
+            col_mapping={'view_test_udf': 'int_field', 'c1': 'str_field'}
         )
         # We're just trying to test metadata here, so reach "under the covers" and link a fake
         # Label Studio project without validation (so we don't need a real Label Studio server)
         from pixeltable.datatransfer.label_studio import LabelStudioProject
         v.tbl_version_path.tbl_version.link(
             LabelStudioProject(4171780),
-            col_mapping={'str_format': 'str_format'}
+            col_mapping={'view_function_call': 'str_format'}
         )
 
+    def __add_expr_columns(self, t: pxt.Table, col_prefix: str, include_expensive_functions=False) -> None:
```

Comment:
Is this missing any exprs (eg, those for which you just filed serialization bugs)?

Reply:
Those aren't serialization bugs, they're bugs that prevent those expressions from being created or turned into a computed column in the first place.

I *believe* that any expression type currently capable of being created and serialized is covered. If you see any that I've overlooked, let me know. As we fix the instantiation bugs, we can include the newly "unlocked" expressions in the db dumps.